### PR TITLE
Fix for departures create task getting stuck

### PR DESCRIPTION
**Pull request by danieldunderfelt**
I noticed that the departures task would get stuck if the user opened a departure-based report while departures were being preloaded. This is a fix for that.

File it under bugfixes https://github.com/HSLdevcom/bultti/issues/148

The fix involves using the cache Set functions to check if the task is running in another process, and if it is, await the result from the cache using timeout polling.

### CODE REVIEW

File: src/departure/common/departureUtils.ts [hunk 1]
```diff
@@ -165,6 +171,20 @@ export async function runDepartureRedisTaskForInspection<ReturnType>(
     return result || undefined
   }
 
+  // If the task has already been started in another process, await the result
+  if (await cacheSetHas(SET_ID, taskId)) {
+    console.log('Thank you Mario! But our task is running in another process!')
```
Comment (Divinare) [inline src/departure/common/departureUtils.ts:176]: Maybe I would log "Task has already started by another process. Waiting for it to finish"

Comment (danieldunderfelt) [reply src/departure/common/departureUtils.ts:176]: Yes, maybe.

That's way less fun though.

Comment (Divinare) [reply src/departure/common/departureUtils.ts]: Yeah I agree. That way seeing this message is less annoying

**Review by Divinare — APPROVED** [review]
Seems to work nicely now!

**Pull request merged**